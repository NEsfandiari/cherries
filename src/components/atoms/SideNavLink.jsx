import React, { Component } from 'react'
import styled from 'styled-components'
import { NavLink } from '../atoms'

const Container = styled.div`
  ul.side-navbar-dropdown {
    list-style: none;
  }

  .side-navbar-dropdown-link {
    margin-left: 25px;
    margin-top: 10px;
    a {
      font-size: 0.8rem;
    }
  }

  li.side-navbar-dropdown-link > a,
  a:visited {
    color: #47525e;
  }
`
class SideNavLink extends Component {
  constructor(props) {
    super(props)
    this.state = {
      display: false,
    }
    this.toggleMenu = this.toggleMenu.bind(this)
  }

  //when links are clicked state will change and dropdown will be shown
  toggleMenu(evt) {
    evt.preventDefault()
    this.setState(st => ({
      display: !st.display,
    }))
  }

  //render NavLinks - dropdown menu will be a ul that renders different li's depending on props passed down from SideNav
  render() {
    const display = this.state.display ? 'initial' : 'none'
    return (
      <Container>
        <li key={this.props.name} className="side-navbar-link">
          <NavLink onClick={this.toggleMenu}>
            {this.props.name} {!this.state.display ? <>+</> : <>-</>}
          </NavLink>
          <ul className="side-navbar-dropdown" style={{ display: display }}>
            {this.props.dropDown.map((li, i) => {
              return (
                <li key={li.name} className="side-navbar-dropdown-link">
                  <NavLink to={`/catalog?search=${li.name}`}>{li.name}</NavLink>
                </li>
              )
            })}
          </ul>
        </li>
      </Container>
    )
  }
}

export default SideNavLink
