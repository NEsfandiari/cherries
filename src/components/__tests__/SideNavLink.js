import React from 'react'
import renderer from 'react-test-renderer'
import SideNavLink from '../atoms/SideNavLink'

describe('SideNavLink', () => {
  it('renders correctly', () => {
    const dropDown = [
      {
        name: 'Basics',
        route: 'basics',
      },
      {
        name: 'Minimalist',
        route: 'minimalist',
      },
      {
        name: 'Girlboss',
        route: 'girlboss',
      },
      {
        name: 'Formal',
        route: 'formal',
      },
      {
        name: 'Boho',
        route: 'boho',
      },
      {
        name: 'Fun',
        route: 'fun',
      },
    ]
    const tree = renderer.create(<SideNavLink dropDown={dropDown} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
