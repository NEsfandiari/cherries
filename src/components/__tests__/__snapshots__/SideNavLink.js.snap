// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavLink renders correctly 1`] = `
<div
  className="sc-VigVT hxFqYU"
>
  <li
    className="side-navbar-link"
  >
    <a
      className="sc-bxivhb dToeqL"
      onClick={[Function]}
    >
       
      +
    </a>
    <ul
      className="side-navbar-dropdown"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <li
        className="side-navbar-dropdown-link"
      >
        <a
          className="sc-bxivhb dToeqL"
          href="basics"
        >
          Basics
        </a>
      </li>
      <li
        className="side-navbar-dropdown-link"
      >
        <a
          className="sc-bxivhb dToeqL"
          href="minimalist"
        >
          Minimalist
        </a>
      </li>
      <li
        className="side-navbar-dropdown-link"
      >
        <a
          className="sc-bxivhb dToeqL"
          href="girlboss"
        >
          Girlboss
        </a>
      </li>
      <li
        className="side-navbar-dropdown-link"
      >
        <a
          className="sc-bxivhb dToeqL"
          href="formal"
        >
          Formal
        </a>
      </li>
      <li
        className="side-navbar-dropdown-link"
      >
        <a
          className="sc-bxivhb dToeqL"
          href="boho"
        >
          Boho
        </a>
      </li>
      <li
        className="side-navbar-dropdown-link"
      >
        <a
          className="sc-bxivhb dToeqL"
          href="fun"
        >
          Fun
        </a>
      </li>
    </ul>
  </li>
</div>
`;
